src/clj/puppetlabs/services/config/puppet_server_config_core.clj

```diff
@@ -92,6 +93,18 @@
                     "conflict with the values to be "
                     "read from puppet: " key-conflicts))))))
 
+(defn check-master-url-prefix
+  "Ensure that Puppet Server's route for the puppet master matches what is in
+  puppet.conf."
+  [webrouter-config puppet-config]
+  (let [puppet-prefix (:master-url-prefix puppet-config)
+        webrouter-prefix (:puppetlabs.services.master.master-service/master-service webrouter-config)]
+    (when-not (= puppet-prefix webrouter-prefix)
+      (throw (Exception.
```

Comment:
We should probably choose a more specific exception type.  We could use `IllegalArgumentException` or `IllegalStateException`, or, we could use this as an opportunity to introduce our own exception base class like `PuppetServerException` if folks want to start trying to head in that direction.  Previously I thought we (PL) had been trying to avoid introducing new Java exception types, but I feel like I've been seeing them start to pop up in some of the other projects related to console services, so maybe it'd be OK.